pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedPrivateMethodRule.java

```diff
@@ -70,7 +70,6 @@ public Object visit(ASTCompilationUnit file, Object param) {
                 .filter(it -> it.getVisibility() == Visibility.V_PRIVATE)
                 .filter(it -> !hasIgnoredAnnotation(it)
                         && !hasExcludedName(it) 
-                        && !it.isAnnotationPresent(Override.class)
```

Comment:
Is this correct, that this check is removed?
Ah, I think I understand: a private method can never override another method - the overridden method has always at least package-private visibility. Adding a Override annotation would be a compile error.

Reply:
Exactly!